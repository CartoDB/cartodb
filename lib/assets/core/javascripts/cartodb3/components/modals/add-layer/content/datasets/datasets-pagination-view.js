var CoreView = require('backbone/core-view');
var PaginationModel = require('../../../../pagination/pagination-model');
var PaginationView = require('../../../../pagination/pagination-view');

/**
 * Responsible for the datasets paginator
 *  ___________________________________________________________________________
 * |                                                                           |
 * |                                             Page 2 of 42 [1] 2 [3][4][5]  |
 * |___________________________________________________________________________|
 *
 */

module.exports = CoreView.extend({
  className: 'DatasetsPaginator',

  initialize: function (opts) {
    if (!opts.routerModel) throw new TypeError('routerModel is required');
    if (!opts.tablesCollection) throw new TypeError('tablesCollection is required');

    this._routerModel = opts.routerModel;
    this._tablesCollection = opts.tablesCollection;

    this.model = new PaginationModel({
      current_page: this._routerModel.get('page')
    });

    this._initBinds();
    this._initViews();
  },

  render: function () {
    this.clearSubViews();
    this.$el.append(this.paginationView.render().el);
    return this;
  },

  _initBinds: function () {
    this.model.bind('change', this.render, this);
    this.model.bind('change:current_page', function () {
      this._routerModel.set('page', this.model.get('current_page'));
    }, this);
    this._tablesCollection.bind('sync', this._updatePaginationModelByCollection, this);
    this._routerModel.bind('change:page', this._updatePaginationModelByRouterModel, this);

    this.add_related_model(this._routerModel);
    this.add_related_model(this._tablesCollection);
    this.add_related_model(this.model);
  },

  _initViews: function () {
    this.paginationView = new PaginationView({
      model: this.model
    });
    this.addView(this.paginationView);
  },

  _updatePaginationModelByCollection: function () {
    this.model.set({
      per_page: this._tablesCollection.getDefaultParam('per_page'),
      total_count: this._tablesCollection.getTotalStat('total_entries')
    });
  },

  _updatePaginationModelByRouterModel: function () {
    this.model.set('current_page', this._routerModel.get('page'));
  }

});
