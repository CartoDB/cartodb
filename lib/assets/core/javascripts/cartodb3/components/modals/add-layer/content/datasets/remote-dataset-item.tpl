<div class="DatasetsList-itemCategory is--<%- isRaster ? 'raster' : geometryType %>Dataset">
  <i data-title="Public dataset" class="CDB-IconFont CDB-IconFont-book DatasetsList-itemStatus <%- canImportDataset ? 'is-public' : 'is-banned' %>"></i>
</div>
<div class="ModalDataset-itemInfo">
  <div class="ModalDataset-itemInfoTitle">
    <h3 class="CDB-Text CDB-Size-large u-bSpace u-ellipsis"><%- title %></h3>
    <% if (description && description.length > 0) { %>
      <p class="u-ellipsis CDB-Text CDB-Size-medium u-altTextColor" title="<%- description %>"><%- description %></p>
    <% } else { %>
      <span class="NoResults CDB-Text CDB-Size-medium"><%- _t('components.modals.add-layer.datasets.item.no-description')%></span>
    <% } %>
  </div>
  <div>
    <ul class="DatasetsList-itemMeta CDB-Text CDB-Size-small u-altTextColor">
      <% if (rowCount) { %>
        <li class="RowsIndicator">
          <%- rowCountFormatted %> <%- _t('components.modals.add-layer.datasets.item.rows-pluralize', { smart_count: rowCount }) %>
        </li>
      <% } %>
      <% if (datasetSize) { %>
        <li class="SizeIndicator">
          <%- datasetSize %>
        </li>
      <% } %>
      <li class="DatasetsList-itemTimeDiff DefaultTimeDiff CDB-Text CDB-Size-small u-altTextColor">
        <%- timeDiff %> <span class="DatasetsList-itemSource js-source"><% if (source) { %><%- _t('components.modals.add-layer.datasets.item.from') %> <%= cdb.core.sanitize.html(source) %><% } %></span>
      </li>
    </ul>
    <div class="DatasetsList-itemMeta DatasetsList-itemTags">
      <% if (tagsCount > 0) { %>
        <div class="DefaultTags CDB-Text CDB-Size-small">
          <% for (var i = 0, l = Math.min(maxTagsToShow, tags.length); i < l; ++i) { %>
            <button class="CDB-Text CDB-Size-small u-upperCase DefaultTags-item js-tag-link u-actionTextColor" value="<%- tags[i] %>"><%- tags[i] %></button><% if (i !== (l-1)) { %><% } %>
          <% } %>
          <% if (tagsCount > maxTagsToShow) { %>
            <%- _t('components.modals.add-layer.datasets.item.tags-more', { tagsCount: tagsCount - maxTagsToShow }) %>
          <% } %>
        </div>
      <% } else { %>
        <span class="NoResults CDB-Text CDB-Size-small u-altTextColor"><%- _t('components.modals.add-layer.datasets.item.no-tags') %></span>
      <% } %>
    </div>
  </div>
</div>
