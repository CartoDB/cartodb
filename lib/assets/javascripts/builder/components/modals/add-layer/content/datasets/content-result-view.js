var CoreView = require('backbone/core-view');
var randomQuote = require('builder/components/loading/random-quote');

/*
 *  Content result default view
 */

module.exports = CoreView.extend({
  events: {
    'click .js-connect': '_onConnectClick'
  },

  initialize: function (opts) {
    if (!opts.routerModel) throw new Error('routerModel is required');
    if (!opts.tablesCollection) throw new Error('tablesCollection is required');
    if (!opts.userModel) throw new Error('userModel is required');
    if (!opts.template) throw new Error('template is required');

    this._userModel = opts.userModel;
    this._routerModel = opts.routerModel;
    this._tablesCollection = opts.tablesCollection;
    this.template = opts.template;

    this._initBinds();
  },

  render: function () {
    var type = this._routerModel.get('content_type');

    this.$el.html(this.template({
      page: this._routerModel.get('page'),
      tag: this._routerModel.get('tag'),
      q: this._routerModel.get('q'),
      shared: this._routerModel.get('shared'),
      locked: this._routerModel.get('locked'),
      library: this._routerModel.get('library'),
      quote: randomQuote(),
      type: type,
      totalItems: this._tablesCollection.size(),
      totalEntries: this._tablesCollection.getTotalStat('total_entries')
    }));

    return this;
  },

  _initBinds: function () {
    this._routerModel.bind('change', this.render, this);
    this._tablesCollection.bind('sync', this.render, this);
    this.add_related_model(this._routerModel);
    this.add_related_model(this._tablesCollection);
  },

  _onConnectClick: function () {
    this.trigger('connectDataset', this);
  }

});
