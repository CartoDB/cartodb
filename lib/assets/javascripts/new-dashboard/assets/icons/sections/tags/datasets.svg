<svg width="14" height="16" xmlns="http://www.w3.org/2000/svg">
  <path d="M7 .5C3.645.5.25 1.53.25 3.5v9c0 1.97 3.395 3 6.75 3s6.75-1.03 6.75-3v-9c0-1.97-3.395-3-6.75-3zM7 2c3.41 0 5.25 1.065 5.25 1.5C12.25 3.935 10.41 5 7 5S1.75 3.935 1.75 3.5C1.75 3.065 3.59 2 7 2zm0 12c-3.41 0-5.25-1.065-5.25-1.5v-1.05c1.302.692 3.28 1.05 5.25 1.05 1.97 0 3.948-.359 5.25-1.05v1.05c0 .435-1.84 1.5-5.25 1.5zm0-3c-3.41 0-5.25-1.065-5.25-1.5V8.45C3.052 9.142 5.03 9.5 7 9.5c1.97 0 3.948-.359 5.25-1.05V9.5c0 .435-1.84 1.5-5.25 1.5zm0-3C3.59 8 1.75 6.935 1.75 6.5V5.45C3.052 6.141 5.03 6.5 7 6.5c1.97 0 3.948-.359 5.25-1.05V6.5C12.25 6.935 10.41 8 7 8z" fill="#2E3C43" fill-rule="nonzero"/>
</svg>
