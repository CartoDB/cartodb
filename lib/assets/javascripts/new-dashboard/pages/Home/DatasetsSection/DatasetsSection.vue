<template>
  <section class="datasets-section">
    <div class="container grid">
      <div class="full-width">
        <SectionTitle class="grid-cell" :title="title">
          <template slot="icon">
            <img src="../../../assets/icons/section-title/data.svg">
          </template>
          <template slot="dropdownButton">
            <SettingsDropdown
              section="datasets"
              :filter="appliedFilter"
              :order="appliedOrder"
              :orderDirection="appliedOrderDirection"
              :metadata="metadata"
              @filterChanged="applyFilter"
              @orderChanged="applyOrder">
              <img svg-inline src="../../../assets/icons/common/filter.svg">
            </SettingsDropdown>
          </template>
          <template slot="actionButton" v-if="!isInitialState">
            <CreateButton visualizationType="dataset">{{ $t(`DataPage.createDataset`) }}</CreateButton>
          </template>
        </SectionTitle>

        <InitialState :title="$t(`DataPage.zeroCase.title`)" v-if="isInitialState">
          <template slot="icon">
            <img svg-inline src="../../../assets/icons/datasets/initialState.svg">
          </template>
          <template slot="description">
            <p class="text is-caption is-txtGrey" v-html="$t(`DataPage.zeroCase.description`)"></p>
          </template>
          <template slot="actionButton">
            <CreateButton visualizationType="maps">{{ $t(`DataPage.zeroCase.createDataset`) }}</CreateButton>
          </template>
        </InitialState>

        <EmptyState v-if="isEmptyState" :text="$t('DataPage.emptyState')" >
          <img svg-inline src="../../../assets/icons/common/compass.svg">
        </EmptyState>

        <DatasetsList
          v-if="numResults"
          :isFetchingDatasets="isFetchingDatasets"
          :datasets="datasets"
          :appliedOrder="appliedOrder"
          :appliedOrderDirection="appliedOrderDirection"
          @applyOrder="applyOrder">
        </DatasetsList>

        <ul v-if="isFetchingDatasets" class="grid-cell grid-cell--col12">
          <li v-for="n in 8" :key="n" class="dataset-item">
            <DatasetCardFake></DatasetCardfake>
          </li>
        </ul>

        <router-link :to="{ name: 'datasets' }" class="title is-small go-to-datasets" v-if="!isInitialState">
          {{ $t('HomePage.DatasetsSection.viewAll') }}
        </router-link>
      </div>
    </div>
  </section>
</template>

<script>
import { mapState } from 'vuex';
import SectionTitle from 'new-dashboard/components/SectionTitle';
import InitialState from 'new-dashboard/components/States/InitialState';
import EmptyState from 'new-dashboard/components/States/EmptyState';
import CreateButton from 'new-dashboard/components/CreateButton';
import DatasetCardFake from 'new-dashboard/components/Dataset/DatasetCardFake';
import DatasetsList from './DatasetList';
import SettingsDropdown from 'new-dashboard/components/Settings/Settings';

export default {
  name: 'DatasetsSection',
  components: {
    CreateButton,
    DatasetCardFake,
    DatasetsList,
    EmptyState,
    InitialState,
    SectionTitle,
    SettingsDropdown
  },
  created: function () {
    this.$store.dispatch('datasets/setResultsPerPage', 8);
    this.fetchDatasets();
  },
  computed: {
    ...mapState({
      isFetchingDatasets: state => state.datasets.isFetching,
      datasets: state => state.datasets.list,
      metadata: state => state.datasets.metadata,
      numResults: state => state.datasets.metadata.total_entries,
      appliedFilter: state => state.datasets.filterType,
      appliedOrder: state => state.datasets.order,
      appliedOrderDirection: state => state.datasets.orderDirection,
      totalUserEntries: state => state.datasets.metadata.total_user_entries
    }),
    title () {
      return this.$t(`DataPage.header.title['${this.appliedFilter}']`);
    },
    isEmptyState () {
      return !this.isFetchingDatasets && !this.numResults && (!this.hasFilterApplied('mine') || this.totalUserEntries > 0);
    },
    isInitialState () {
      return !this.isFetchingDatasets && this.hasFilterApplied('mine') && this.totalUserEntries <= 0;
    }
  },
  methods: {
    applyOrder (orderOptions) {
      this.$store.dispatch('datasets/orderDatasets', orderOptions);
    },
    applyFilter (filterType) {
      this.$store.dispatch('datasets/filterDatasets', filterType);
    },
    fetchDatasets () {
      this.$store.dispatch('datasets/fetchDatasets');
    },
    hasFilterApplied (filter) {
      return this.appliedFilter === filter;
    }
  }
};
</script>

<style scoped lang="scss">
@import "stylesheets/new-dashboard/variables";

.datasets-section {
  .head-section,
  .empty-state {
    width: 100%;
  }
}

.go-to-datasets {
  display: block;
  padding-top: 64px;
  letter-spacing: 1px;
  text-align: center;
}

.full-width {
  width: 100%;
}
</style>
