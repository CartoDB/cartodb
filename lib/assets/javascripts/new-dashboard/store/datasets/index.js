import CartoNode from 'carto-node';
import toObject from 'new-dashboard/utils/to-object';
import Filters, { defaultParams as filtersDefaultParams } from 'new-dashboard/core/filters';

const client = new CartoNode.AuthenticatedClient();

const DEFAULT_VALUES = {
  filter: 'mine',
  order: 'updated_at',
  orderDirection: 'desc'
};

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: DEFAULT_VALUES.filter,
    order: DEFAULT_VALUES.order,
    orderDirection: DEFAULT_VALUES.orderDirection,
    list: {},
    metadata: {},
    page: 1,
    numPages: 1,
    resultsPerPage: 1
  },
  mutations: {
    setRequestError (state, error) {
      state.isFetching = false;
      state.isErrored = true;
      state.error = error;
    },

    setDatasets (state, datasets) {
      state.list = toObject(datasets.visualizations, 'id');
      state.metadata = {
        total_entries: datasets.total_entries,
        total_likes: datasets.total_likes,
        total_shared: datasets.total_shared,
        total_locked: datasets.total_locked,
        total_user_entries: datasets.total_user_entries
      };

      state.isFetching = false;
    },

    setFetchingState (state) {
      state.isFetching = true;
      state.isErrored = false;
      state.error = {};
    },

    setDatasetAttributes (state, {datasetId, datasetAttributes}) {
      Object.assign(state.list[datasetId], datasetAttributes);
    },

    setPagination (state, page = 1) {
      state.page = page;
      state.numPages = Math.ceil(state.metadata.total_entries / filtersDefaultParams.per_page) || 1;
    },

    setFilterType (state, filterType = 'mine') {
      state.filterType = filterType;
    },

    setOrder (state, orderOptions) {
      state.order = orderOptions.order || DEFAULT_VALUES.order;
      state.orderDirection = orderOptions.direction || DEFAULT_VALUES.orderDirection;
    },

    setResultsPerPage (state, resultsPerPage) {
      state.resultsPerPage = resultsPerPage;
    },

    updateLike (state, {datasetId, liked}) {
      state.list[datasetId].liked = liked;
    },

    updateNumberLikes (state, {datasetId, likes}) {
      state.list[datasetId].likes = likes;
    }
  },
  actions: {
    fetchDatasets (context) {
      context.commit('setFetchingState');

      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order,
        order_direction: context.state.orderDirection,
        per_page: context.state.resultsPerPage,
        with_dependent_visualizations: 10
      };

      client.getVisualization('',
        params,
        function (err, _, data) {
          if (err) {
            context.commit('setRequestError', err);
            return;
          }
          context.commit('setDatasets', data);
          context.commit('setPagination', context.state.page);
        });
    },

    filterDatasets (context, filter) {
      context.commit('setPagination', 1);
      context.commit('setFilterType', filter);
      context.dispatch('fetchDatasets');
    },

    orderDatasets (context, orderOptions) {
      context.commit('setPagination', 1);
      context.commit('setOrder', orderOptions);
      context.dispatch('fetchDatasets');
    },

    like (context, dataset) {
      const currentLikeStatus = dataset.liked;
      context.commit('updateLike', { datasetId: dataset.id, liked: true });
      client.like(dataset.id,
        function (err, _, data) {
          if (err) {
            context.commit('updateLike', { datasetId: dataset.id, liked: currentLikeStatus });
            return;
          }
          context.commit('updateNumberLikes', { datasetId: dataset.id, likes: data.likes });
        }
      );
    },

    deleteLike (context, dataset) {
      const currentLikeStatus = dataset.liked;
      context.commit('updateLike', { datasetId: dataset.id, liked: false });
      client.deleteLike(dataset.id,
        function (err, _, data) {
          if (err) {
            context.commit('updateLike', { datasetId: dataset.id, liked: currentLikeStatus });
            return;
          }
          context.commit('updateNumberLikes', { datasetId: dataset.id, likes: data.likes });
        }
      );
    },

    updateDataset (context, datasetOptions) {
      context.commit('setDatasetAttributes', datasetOptions);
    },

    setResultsPerPage (context, perPage) {
      context.commit('setResultsPerPage', perPage);
    },

    setURLOptions (context, options) {
      context.commit('setPagination', options.page);
      context.commit('setFilterType', options.filter);
      context.commit('setOrder', { order: options.order, direction: options.order_direction });
      context.dispatch('fetchDatasets');
    },

    resetFilters (context) {
      context.commit('setPagination', 1);
      context.commit('setFilterType', 'mine');
      context.commit('setResultsPerPage', 12);
      context.commit('setOrder', { order: DEFAULT_VALUES.order, direction: DEFAULT_VALUES.orderDirection });
    }
  }
};

export default datasets;
