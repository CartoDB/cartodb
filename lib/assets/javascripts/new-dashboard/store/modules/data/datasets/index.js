import Filters from 'new-dashboard/core/configuration/filters';

import * as VisualizationActions from '../../../actions/visualizations';
import * as VisualizationMutations from '../../../mutations/visualizations';

const DEFAULT_VALUES = {
  filter: 'mine',
  order: 'favorited,updated_at',
  orderDirection: 'desc,desc'
};

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: DEFAULT_VALUES.filter,
    order: DEFAULT_VALUES.order,
    orderDirection: DEFAULT_VALUES.orderDirection,
    list: {},
    metadata: {},
    page: 1,
    numPages: 1,
    resultsPerPage: 1
  },
  mutations: {
    setFetchingState: VisualizationMutations.setFetchingState,
    setFilterType: VisualizationMutations.setFilterType,
    setOrder: VisualizationMutations.setOrder(DEFAULT_VALUES),
    setPagination: VisualizationMutations.setPagination,
    setResultsPerPage: VisualizationMutations.setResultsPerPage,
    setRequestError: VisualizationMutations.setRequestError,
    setVisualizations: VisualizationMutations.setVisualizations,
    updateNumberLikes: VisualizationMutations.updateNumberLikes,
    updateVisualization: VisualizationMutations.updateVisualization
  },
  actions: {
    deleteLike: VisualizationActions.deleteLike,
    filter: VisualizationActions.filter,
    like: VisualizationActions.like,
    order: VisualizationActions.order,
    resetFilters: VisualizationActions.resetFilters(DEFAULT_VALUES),
    setResultsPerPage: VisualizationActions.setResultsPerPage,
    setURLOptions: VisualizationActions.setURLOptions,
    updateVisualization: VisualizationActions.updateVisualization,

    fetch (context) {
      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order,
        order_direction: context.state.orderDirection,
        per_page: context.state.resultsPerPage,
        with_dependent_visualizations: 10
      };

      VisualizationActions.fetchVisualizations(context, params);
    }
  }
};

export default datasets;
