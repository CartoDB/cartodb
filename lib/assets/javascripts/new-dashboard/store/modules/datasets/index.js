import toObject from 'new-dashboard/utils/to-object';
import Filters, { defaultParams as filtersDefaultParams } from 'new-dashboard/core/configuration/filters';

import * as VisualizationActions from '../../actions/visualizations';
import * as VisualizationMutations from '../../mutations/visualizations';

const DEFAULT_VALUES = {
  filter: 'mine',
  order: 'updated_at',
  orderDirection: 'desc'
};

const datasets = {
  namespaced: true,
  state: {
    isFetching: false,
    isFiltered: false,
    isErrored: false,
    error: {},
    filterType: DEFAULT_VALUES.filter,
    order: DEFAULT_VALUES.order,
    orderDirection: DEFAULT_VALUES.orderDirection,
    list: {},
    metadata: {},
    page: 1,
    numPages: 1,
    resultsPerPage: 1
  },
  mutations: {
    setRequestError (state, error) {
      state.isFetching = false;
      state.isErrored = true;
      state.error = error;
    },

    setVisualizations (state, datasets) {
      state.list = toObject(datasets.visualizations, 'id');
      state.metadata = {
        total_entries: datasets.total_entries,
        total_likes: datasets.total_likes,
        total_shared: datasets.total_shared,
        total_locked: datasets.total_locked,
        total_user_entries: datasets.total_user_entries
      };

      state.isFetching = false;
    },

    setFetchingState (state) {
      state.isFetching = true;
      state.isErrored = false;
      state.error = {};
    },

    setPagination (state, page = 1) {
      state.page = page;
      state.numPages = Math.ceil(state.metadata.total_entries / filtersDefaultParams.per_page) || 1;
    },

    setFilterType (state, filterType = 'mine') {
      state.filterType = filterType;
    },

    setOrder (state, orderOptions) {
      state.order = orderOptions.order || DEFAULT_VALUES.order;
      state.orderDirection = orderOptions.direction || DEFAULT_VALUES.orderDirection;
    },

    setResultsPerPage (state, resultsPerPage) {
      state.resultsPerPage = resultsPerPage;
    },

    like: VisualizationMutations.updateLike,
    updateNumberLikes: VisualizationMutations.updateNumberLikes
  },
  actions: {
    fetch (context) {
      const params = {
        ...Filters[context.state.filterType],
        types: 'table',
        page: context.state.page,
        order: context.state.order,
        order_direction: context.state.orderDirection,
        per_page: context.state.resultsPerPage,
        with_dependent_visualizations: 10
      };

      VisualizationActions.fetchVisualizations(context, params);
    },

    filterDatasets (context, filter) {
      context.commit('setPagination', 1);
      context.commit('setFilterType', filter);
      context.dispatch('fetch');
    },

    orderDatasets (context, orderOptions) {
      context.commit('setPagination', 1);
      context.commit('setOrder', orderOptions);
      context.dispatch('fetch');
    },

    like: VisualizationActions.like,
    deleteLike: VisualizationActions.deleteLike,
    updateVisualization: VisualizationActions.updateVisualization,

    setResultsPerPage (context, perPage) {
      context.commit('setResultsPerPage', perPage);
    },

    setURLOptions (context, options) {
      context.commit('setPagination', options.page);
      context.commit('setFilterType', options.filter);
      context.commit('setOrder', { order: options.order, direction: options.order_direction });
      context.dispatch('fetch');
    },

    resetFilters (context) {
      context.commit('setPagination', 1);
      context.commit('setFilterType', 'mine');
      context.commit('setResultsPerPage', 12);
      context.commit('setOrder', { order: DEFAULT_VALUES.order, direction: DEFAULT_VALUES.orderDirection });
    }
  }
};

export default datasets;
