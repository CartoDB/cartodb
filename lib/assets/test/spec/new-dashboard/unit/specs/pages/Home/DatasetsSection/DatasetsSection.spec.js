import { createLocalVue, shallowMount } from '@vue/test-utils';
import Vuex from 'vuex';
import datasets from '../../../fixtures/datasets';
import DatasetsSection from 'new-dashboard/pages/Home/DatasetsSection/DatasetsSection';

jest.mock('new-dashboard/components/CreateButton', () => {});

const $t = key => key;

const localVue = createLocalVue();
localVue.use(Vuex);

describe('DatasetsSection.vue', () => {
  it('should render properly', () => {
    const store = createStore();
    const datasetsSection = createDatasetsSection(store);
    expect(datasetsSection).toMatchSnapshot();
  });

  it('should render initial state', () => {
    const store = createStore({
      datasets: {
        isFetchingDatasets: false,
        list: datasets.visualizations,
        metadata: {
          total_user_entries: 0
        },
        filterType: 'mine'
      }
    });

    const datasetsSection = createDatasetsSection(store);
    expect(datasetsSection).toMatchSnapshot();
  });

  it('should render emptyState', () => {
    const store = createStore({
      datasets: {
        isFetchingDatasets: false,
        datasets: datasets.visualizations,
        metadata: {
          total_entries: 0
        },
        filterType: 'locked'
      }
    });
    const datasetsSection = createDatasetsSection(store);
    expect(datasetsSection).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('applyOrder: should dispatch datasets/orderDatasets with order options', () => {
      const store = createStore();
      spyOn(store, 'dispatch');

      const datasetsSection = createDatasetsSection(store);

      datasetsSection.vm.applyOrder({ order: 'updated_at', direction: 'desc' });

      expect(store.dispatch).toHaveBeenCalledWith('datasets/orderDatasets', {
        order: 'updated_at',
        direction: 'desc'
      });
    });

    it('applyFilter: should dispatch datasets/filterDatasets with filter type', () => {
      const store = createStore();
      spyOn(store, 'dispatch');

      const datasetsSection = createDatasetsSection(store);

      datasetsSection.vm.applyFilter('locked');

      expect(store.dispatch).toHaveBeenCalledWith('datasets/filterDatasets', 'locked');
    });

    it('fetchDatasets: should dispatch datasets/fetchDatasets', () => {
      const store = createStore();
      spyOn(store, 'dispatch');

      const datasetsSection = createDatasetsSection(store);

      datasetsSection.vm.fetchDatasets();

      expect(store.dispatch).toHaveBeenCalledWith('datasets/fetchDatasets');
    });

    it('hasFilterApplied: should return true if applied filter is equal to the one passed', () => {
      const store = createStore();
      const datasetsSection = createDatasetsSection(store);

      expect(datasetsSection.vm.hasFilterApplied('mine')).toBe(true);
    });
  });
});

function createStore (customStoreData) {
  const storeData = {
    datasets: {
      isFetchingDatasets: false,
      datasets: datasets.visualizations,
      metadata: {
        total_user_entries: 2,
        total_entries: 2
      },
      filterType: 'mine',
      order: 'updated_at',
      orderDirection: 'desc'
    }
  };

  let storeInstance = new Vuex.Store({
    state: customStoreData || storeData
  });

  return storeInstance;
}

function createDatasetsSection (store) {
  return shallowMount(DatasetsSection, {
    mocks: { $t },
    store,
    localVue
  });
}
