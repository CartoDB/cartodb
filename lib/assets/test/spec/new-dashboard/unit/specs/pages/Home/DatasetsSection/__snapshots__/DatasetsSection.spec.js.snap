// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetsSection.vue should render emptyState 1`] = `
<section class="datasets-section">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub title="DataPage.header.title['locked']" class="grid-cell"><template></template> <template></template> <template></template></sectiontitle-stub>
      <!---->
      <emptystate-stub text="DataPage.emptyState"><img svg-inline="" src="../../../assets/icons/common/compass.svg"></emptystate-stub>
      <!---->
      <!---->
      <router-link to="[object Object]" class="title is-small go-to-datasets">
        HomePage.DatasetsSection.viewAll
      </router-link>
    </div>
  </div>
</section>
`;

exports[`DatasetsSection.vue should render initial state 1`] = `
<section class="datasets-section">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub title="DataPage.header.title['mine']" class="grid-cell"><template></template> <template></template>
        <!---->
      </sectiontitle-stub>
      <initialstate-stub title="DataPage.zeroCase.title"><template></template> <template></template> <template></template></initialstate-stub>
      <!---->
      <!---->
      <!---->
      <!---->
    </div>
  </div>
</section>
`;

exports[`DatasetsSection.vue should render properly 1`] = `
<section class="datasets-section">
  <div class="container grid">
    <div class="full-width">
      <sectiontitle-stub title="DataPage.header.title['mine']" class="grid-cell"><template></template> <template></template> <template></template></sectiontitle-stub>
      <!---->
      <!---->
      <datasetslist-stub appliedorder="updated_at" appliedorderdirection="desc"></datasetslist-stub>
      <!---->
      <router-link to="[object Object]" class="title is-small go-to-datasets">
        HomePage.DatasetsSection.viewAll
      </router-link>
    </div>
  </div>
</section>
`;
